### PR TITLE
Case insensitive chain name

### LINKED ISSUE #526: Make the name of ChainConfig case insensitive
## Feature description 

Currently, the name of the chain config is not In `x/profile` is not specified to use lower case. 
This means that currently it's possible to have many chain configs with very similar chain name, say:

* Cosmos
* cosmos
* cosMos
* ...  

This might be a problem in the future due to the possible impersonification.

## Implementation proposal

To solve it, we can convert the name to lower case in the constructor and make sure it is all in lower case by Validate().
```golang
// NewChainConfig allows to build a new ChainConfig instance
func NewChainConfig(name string) ChainConfig {
	return ChainConfig{
		Name: strings.ToLower(name),
	}
}

// Validate checks the validity of the ChainConfig
func (c ChainConfig) Validate() error {
	if strings.TrimSpace(c.Name) == "" {
		return fmt.Errorf("chain name cannot be empty or blank")
	}
	if strings.ToLower(c.Name) != c.Name {
	        return fmt.Errorof("chain name should be lower case")
	}
	return nil
}
```

What do you think? @RiccardoM @bragaz

### PR DESCRIPTION
**Pull request by dadamu**
## Description

This PR makes the name of `ChainConfig` case insensitive.
Closes #526

## Checklist
- [x] Targeted PR against correct branch.
- [x] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote unit tests.
- [ ] Wrote integration tests (simulation & CLI). 
- [ ] Updated the documentation. 
- [ ] Added an entry to the `CHANGELOG.md` file.
- [ ] Re-reviewed `Files changed` in the Github PR explorer.

### CODE REVIEW

**Review by bragaz — APPROVED** [review]

File: x/profiles/types/models_chain_links.go [hunk 1]
```diff
@@ -22,7 +22,7 @@ import (
 // NewChainConfig allows to build a new ChainConfig instance
 func NewChainConfig(name string) ChainConfig {
 	return ChainConfig{
-		Name: name,
+		Name: strings.ToLower(name),
```
Comment (RiccardoM) [inline x/profiles/types/models_chain_links.go:25]: I think this should be removed if we want the `Validate` method to fail

**Review by RiccardoM — COMMENTED** [review]
The implementation looks good, but I'm having a hard time deciding whether this is worth or it might have some backslash in the future

**Comment by bragaz** [comment]
@RiccardoM why backslash? I don't think we will see chains where name differs only for some case letter.

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,6 @@
+# Unreleased
+- Made the name of ChainConfig case-insensitive ([#533](https://github.com/desmos-labs/desmos/pull/533))
```
Comment (RiccardoM) [inline CHANGELOG.md:2]:
```suggestion
- Require chain name to be lowercase ([#533](https://github.com/desmos-labs/desmos/pull/533))
```

**Review by RiccardoM — APPROVED** [review]

**Pull request merged**